From b102d027a00d93add88595d486172a50ba182a33 Mon Sep 17 00:00:00 2001
From: Marius Gripsgard <marius@ubports.com>
Date: Wed, 25 Mar 2020 03:34:36 +0100
Subject: [PATCH] [sim] Make sure we send change notifacation for pin values

This makes sure we send change notifcations if the pin values get set to
none after sim_inserted_nofify has been called.
---
 src/sim.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/sim.c b/src/sim.c
index 535ccbcaf..f8b86a175 100644
--- a/src/sim.c
+++ b/src/sim.c
@@ -80,6 +80,7 @@ struct ofono_sim {
 	gboolean locked_pins[OFONO_SIM_PASSWORD_SIM_PUK]; /* Number of PINs */
 
 	int pin_retries[OFONO_SIM_PASSWORD_INVALID];
+	bool pin_values_dirty;
 
 	enum ofono_sim_phase phase;
 	unsigned char mnc_length;
@@ -2925,6 +2926,8 @@ void ofono_sim_inserted_notify(struct ofono_sim *sim, ofono_bool_t inserted)
 		sim->pin_retries[OFONO_SIM_PASSWORD_SIM_PIN2] = -1;
 		sim->pin_retries[OFONO_SIM_PASSWORD_SIM_PUK2] = -1;
 
+		sim->pin_values_dirty = 1;
+
 		pin_cache_remove(sim->iccid);
 
 		sim_free_state(sim);
@@ -3212,8 +3215,9 @@ static void sim_pin_query_cb(const struct ofono_error *error,
 		return;
 	}
 
-	if (sim->pin_type != pin_type) {
+	if (sim->pin_type != pin_type || sim->pin_values_dirty) {
 		sim->pin_type = pin_type;
+		sim->pin_values_dirty = 0;
 
 		if (pin_type != OFONO_SIM_PASSWORD_NONE &&
 				password_is_pin(pin_type) == FALSE)
